docs/package-management.md

```diff
@@ -267,3 +267,129 @@ The advantage of this approach —as opposed to vendoring the JavaScript package
+
+```text
+...
+depends: [
+  ...
+  "melange-fetch" {dev}
+]
+pin-depends: [
+  [ "melange-fetch.dev" "git+https://github.com/melange-community/melange-fetch" ]
+]
+```
+
+Once installed, the package name can be added to the `dune` file:
+
+```text
+(melange.emit
+ (target output)
+ (alias react)
+ (libraries melange lib reason-react melange-fetch)
+ (preprocess
+  (pps reactjs-jsx-ppx))
+ (module_systems es6))
+```
+
+Depending on where the installed package is used will determine in which `dune` file the package should be referenced. In the next section we will see how to reference a package from within a library subfolder.
+
+### NPM packages
+
+A number of Melanage compatible packages can be found on NPM. These packages are typically bindings to JavaScript libraries. Many older, but still useful, compatible BuckleScript libraries can be found on NPM, e.g., `bs-json`. An NPM package is installed like any normal NPM package is: `npm install @glennsl/bs-json`, which will add the package to the `package.json` file at the root of our project.
+
+In order to get dune to pick up and process the NPM package so that it can be consumed in our application, the `dune` file at the root of our project must be updated with a `subdir` stanza:
+
+```text
+(subdir
+ node_modules
+ (vendored_dirs @glennsl)
+ (dirs @glennsl)
+ (subdir
+  @glennsl
+  (subdir
+   bs-json
+   (subdir
+    src
+    (library
+     (name bs_json)
+     (wrapped false)
+     (modes melange))))))
+```
+
+(If the `dune` file contains the line `(dirs :standard \ node_modules)`, it should be removed.)
+Note the `(vendored_dirs @glennsl)` field in the stanza. This is used to silence warnings from vendored libraries, keeping the terminal output relevant to our application code.
+
+In our project structure above we have the file <code class="text-ocaml">data.ml</code><code class="text-reasonml">data.re</code> under the folder `src/lib`. If we want use the `bs-json` package from within the <code class="text-ocaml">data.ml</code><code class="text-reasonml">data.re</code> file then we need to add the package name to the `dune` file in the same folder, i.e., `src/lib/dune`:
+
+```text
+(library
+ (name data)
+ (libraries bs_json)
+ (modes melange))
+```
+
+Note that the package `bs-json` was renamed to `bs_json` in the `subdir` stanza and is referenced as `bs_json` in the `dune` file.
+
+If we install any more NPM packages then a new `subdir` stanza will have to be added to `dune` file in the root of our project and referenced in the `dune` file that lives next the code we want to use the package from. See [this dune file](https://github.com/jchavarri/pupilfirst/blob/b6fe6fa27814336b9e92f965449f3e010e148c4b/dune) for a larger example that uses multiple NPM packages.
+
+**The advantage of publishing packages on opam is that none of the these `subdir` and `vendored_dirs` configurations are necessary.**
```

Comment:
It is fine to include these docs for now as we transition packages to opam. But we should try to remove this section asap cc @davesnx @anmonteiro